week1/1.5/Notes/Notes/Controllers/Model Controllers/NoteController.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  NoteController.swift
+//  Notes
+//
+//  Created by Eric Lanza on 7/6/18.
+//  Copyright © 2018 ETLanza. All rights reserved.
+//
+
+import Foundation
+
+class NoteController: Codable {
+    
+    static let shared = NoteController()
+    
+    var notes = [Note]()
+    
+    //MARK: - CRUD Methods
+    func addNote(_ note: Note) {
+        notes.append(note)
+        saveToPersistentStore()
+    }
+    
+    func updateNote(_ note: Note, withBodyText bodyText: String) {
+        note.bodyText = bodyText
+        saveToPersistentStore()
+    }
+    
+    func deleteNote(_ note: Note) {
+        guard let indexOfNote = NoteController.shared.notes.index(of: note) else { return }
```

Comment:
This function probably shouldn't fail silently. Not a big issue for a simple little test app but if you were to improve this app then it should return something like `Note?` where it returned the deleted note if found and nil otherwise.